Sdk-dapp V5 migration

razvantomegea opened the pull request.

razvantomegea committed:
Merge pull request #66 from multiversx/development

1.0.3

liviuf-ancas committed:
added sovereign (#68)

Co-authored-by: liviuancas-elrond <liviu.ancas@elrond.com>

razvantomegea committed:
Merge pull request #71 from multiversx/development

1.0.4

razvantomegea committed:
Development (#75)

* Merge pull request #73 from multiversx/rt/fix/register-token-contract-address

Fixed sovereign contract address in register token

* v1.0.5 (#74)

* Updated version

* Fixed test and scripts

* fixed script

* Fixed test

arhtudormorar committed:
Fix README (#78)

* Fix README

* Update CHANGELOG

* Update CHANGELOG.md

razvantomegea committed:
Merge pull request #80 from multiversx/development

v1.0.6

arhtudormorar committed:
Merge pull request #88 from multiversx/development

Development

razvantomegea committed:
Sdk-dapp v5 migration

arhtudormorar committed:
Merge pull request #91 from multiversx/development

Development

razvantomegea committed:
Merge pull request #92 from multiversx/development

Edit config wallet addresses

razvantomegea committed:
Refactor SDK imports and update toast handling in AxiosInterceptor

- Removed deprecated `@multiversx/sdk-dapp-core` imports, replacing them with `@multiversx/sdk-dapp` equivalents across various files.
- Updated the toast creation logic in `handleError.ts` to utilize the new `createCustomToast` function.
- Introduced new components in the `sdkDapp` library, including `CopyButton`, `ExplorerLink`, and `FormatAmount`.
- Adjusted the `CreateMnemonics`, `CreateRecoverDownloadScreen`, and other components to use the new imports from `lib/sdkDapp`.
- Updated package dependencies in `package.json` and `yarn.lock` to reflect the latest versions of SDK packages.

razvantomegea committed:
Refactor SignMessage component and related files

- Simplified the SignMessage component by removing unused hooks and state management.
- Updated the handling of message signing, including error and success states.
- Enhanced the SignSuccess and SignFailure components to accept props for better reusability.
- Refactored the decodeMessage helper function to streamline message encoding and decoding processes.
- Improved UI elements for better user experience during message signing.

razvantomegea committed:
Merge branch 'main' into rt/feature/sdk-dapp-v5

razvantomegea committed:
Refactor SDK imports and streamline component structure

- Updated imports across the application to replace `sdkDapp` references with a unified `lib` import.
- Simplified the `App.tsx` structure by removing unnecessary hooks and dispatch calls.
- Enhanced the `Header` component with new `ConnectButton` and `NotificationsButton` components for improved user interaction.
- Refactored various components to utilize the new import structure, ensuring consistency and reducing redundancy.
- Removed deprecated `sdkDappCoreUI` components and adjusted related files accordingly.

razvantomegea committed:
Refactor provider structure and enhance wallet integration

- Introduced new `KeystoreProvider` and `PemProvider` classes to manage wallet interactions.
- Implemented `KeystoreLoginModal` and `PemLoginModal` for user authentication via keystore and PEM files.
- Updated the `provider` object to utilize the new provider classes, improving modularity and maintainability.
- Refactored related components and hooks to integrate with the new provider structure, ensuring a seamless user experience.
- Cleaned up imports and removed deprecated components related to wallet handling.

razvantomegea committed:
Refactor header and network configuration components

- Updated the Header component to use `useCallback` for the logout function, improving performance and error handling.
- Enhanced the ConnectButton component with `useMemo` for better optimization.
- Corrected import references from `sharedNetorks` to `sharedNetworks` across multiple configuration files, ensuring consistency.
- Improved error handling in the getIsInWebview function and adjusted the signature handling in the SignMessage component for better reliability.
- Refactored various components to utilize TextEncoder for encoding messages, enhancing compatibility and performance.

razvantomegea committed:
Update SDK dependencies and refactor Keystore components

- Added `@multiversx/sdk-dapp` version `5.0.0-alpha.7` to `package.json` and updated related dependencies in `yarn.lock`.
- Refactored Keystore components by removing deprecated files and restructuring imports for better organization.
- Updated the `nativeAuth` import path in `useInitToken` to enhance clarity and maintainability.
- Cleaned up unused components and hooks related to the Keystore functionality, streamlining the codebase.

razvantomegea committed:
Merge branch 'development' into rt/feature/sdk-dapp-v5

razvantomegea committed:
Refactor account-related hooks and imports

- Replaced `useGetAccountInfo` with `useGetAccount` across multiple components and hooks for consistency.
- Removed the `sdkDappForm.constants.ts` file and updated related imports to streamline the codebase.
- Enhanced the `KeystoreProvider` and `PemProvider` to return 'custom' in the `getType` method, preparing for future updates.
- Cleaned up unused imports and adjusted routes for better organization and maintainability.

razvantomegea committed:
Enhance component structure and update imports

- Added a new `Loader` component and exported it from the main components index.
- Updated import paths in various components to replace `lib` references with `types` for consistency.
- Refactored the `Button`, `Card`, `MxLink`, and `OutputContainer` components to use the new import structure.
- Introduced a new `TransactionsTable` component to streamline transaction display in the Dashboard.
- Removed deprecated `sdkDappUI` components and adjusted related files accordingly.

razvantomegea committed:
Enhance SDK Dapp hooks and utilities

- Added new hooks: `useGetIsLoggedIn` and `useGetLoginInfo` to improve account management.
- Introduced `useGetPendingTransactions` for better transaction handling.
- Added `stringIsFloat` utility function to enhance number validation capabilities.
- Refactored conditional rendering in the `Sign` component for improved readability.

razvantomegea committed:
Update selectors and refactor HookValidationOutcome component

- Added `loginInfoSelector` to the SDK Dapp selectors for improved account state management.
- Refactored `HookValidationOutcome` to utilize `getAccountProvider` and `loginInfoSelector`, enhancing clarity and maintainability.
- Updated conditional rendering logic to streamline wallet connection handling.

razvantomegea committed:
Remove CopyButton component and related files; refactor hooks to use getAccountProvider for improved account management

arhtudormorar commented on package.json:
```diff
@@ -44,10 +44,11 @@
-    "@multiversx/sdk-dapp-form": "2.0.2",
+    "@multiversx/sdk-core": "14.0.2",
+    "@multiversx/sdk-dapp": "5.0.0-alpha.7",
+    "@multiversx/sdk-dapp-core-ui": "0.0.0-alpha.15",
```
we have stable version here

arhtudormorar commented on src/App.tsx:
```diff
@@ -1,95 +1,42 @@
-    </DappProvider>
+    <Layout>
+      <Routes>
+        <Route path={routeNames.unlock} element={<Unlock />} />
```
unlock as chidl route of home ?

arhtudormorar commented on src/App.tsx:
```diff
@@ -1,95 +1,42 @@
-      </AxiosInterceptorContext.Interceptor>
-    </AxiosInterceptorContext.Provider>
+    <AxiosInterceptor>
+      <BatchTransactionsContextProvider>
```
why is BatchTransactionsContextProvider still needed?

arhtudormorar commented on src/components/AxiosInterceptor/helpers/handleError.ts:
```diff
@@ -1,5 +1,4 @@
-import { faWarning } from '@fortawesome/free-solid-svg-icons';
-import { addNewCustomToast, CustomToastType } from 'lib';
+import { createCustomToast } from 'lib';
```
Module '"lib"' has no exported member 'createCustomToast'.

arhtudormorar commented on src/components/Layout/Header/Header.tsx:
```diff
@@ -1,45 +1,59 @@
+    } finally {
+      navigate(RouteNamesEnum.home);
+    }
+  }, [provider, navigate]);
```
Why use a callback and why add navigate as param

arhtudormorar commented on src/components/Layout/Header/components/ConnectButton.tsx:
```diff
@@ -0,0 +1,23 @@
+
+  const unlockPanelManager = useMemo(
+    () =>
+      UnlockPanelManager.init({
```
This logic should be in unlock page

arhtudormorar commented on src/components/PageState/PageState.tsx:
```diff
@@ -0,0 +1,59 @@
+  iconSize?: SizeProp;
+  iconBgClass?: string;
+  action?: ReactNode;
+  description?: string | ReactNode;
```
I think react node includes string

arhtudormorar commented on src/helpers/app/getIsInWebview.ts:
```diff
@@ -1,12 +1,21 @@
+      return false;
+    }
+
+    const isExternalProvider = !Object.values(ProviderTypeEnum).includes(
```
I think this is no longer needed

arhtudormorar commented on src/helpers/app/provider.ts:
```diff
@@ -1,31 +1,38 @@
 
-export const provider: IDappProvider = {
-  init: async () => {
+class CustomProvider implements IProvider {
```
Rename to FileProvider ?

arhtudormorar commented on src/hooks/tokens/useGetTokensWithEgld.ts:
```diff
@@ -22,8 +28,9 @@ const defaultValues = {
-  const { network } = useGetNetworkConfig();
+  const { address, account } = useGetAccountInfo();
+  const network = networkSelector(getState());
+  const websocketEvent = websocketEventSelector(getState());
```
This might not be reactive. Same for network.

arhtudormorar commented on src/hooks/useIsWebProvider/useIsWebProvider.ts:
```diff
@@ -1,8 +1,9 @@
+  const providerType = provider.getType();
+  const isWebProvider = providerType === ProviderTypeEnum.webview;
 
   return { isWebProvider };
```
No need to return an object

arhtudormorar commented on src/index.tsx:
```diff
@@ -1,7 +1,47 @@
+    name: 'PEM File',
+    type: 'pemProvider',
+    iconUrl: `${window.location.origin}/pem-icon.svg`,
+    constructor: async (options?: any) => new PemProvider(options)
```
No need for any, options are inferred

arhtudormorar commented on src/index.tsx:
```diff
@@ -1,7 +1,47 @@
+    name: 'Keystore File',
+    type: 'keystoreProvider',
+    iconUrl: `${window.location.origin}/keystore-icon.svg`,
+    constructor: async (options?: any) => new KeystoreProvider(options)
```
No need for any

arhtudormorar commented on src/index.tsx:
```diff
@@ -1,7 +1,47 @@
+];
+
+(window as any).multiversx = {};
+(window as any).multiversx.providers = providers;
```
I think it's better to add them to the provider list instead of adding them to the window

arhtudormorar commented on src/index.tsx:
```diff
@@ -1,7 +1,47 @@
+      walletAddress: window.location.origin
+    },
+    successfulToastLifetime: 5000
+  }
```
Provider list should be somewhere here

arhtudormorar commented on src/pages/Dashboard/widgets/Account/Account.tsx:
```diff
@@ -1,24 +1,24 @@
 export const Account = () => {
-  const { network } = useGetNetworkConfig();
-  const { activeNetwork } = useSelector(networkSelector);
+  const activeNetwork = networkSelector(getState());
```
This is not reactive

arhtudormorar commented on src/pages/Dashboard/widgets/Account/Account.tsx:
```diff
@@ -1,24 +1,24 @@
-  const { hasRegisterToken, hasSovereignTransfer } = activeNetwork;
-  const explorerAddress = useSdkDappSelector(explorerAddressSelector);
+
+  // TODO: fix this
```
Yes, check cast

arhtudormorar commented on src/pages/Hook/HookValidationOutcome/HookValidationOutcome.tsx:
```diff
@@ -26,8 +27,9 @@ export const HookValidationOutcome = ({
-  const isWalletConnectV2Initializing = useGetIsWalletConnectV2Initialized();
+  const provider = getAccountProvider();
+  const providerType = provider.getType();
+  const { walletConnectV2Initializing } = loginInfoSelector(getState());
```
Whenever we use get state, it will not be reactive

arhtudormorar commented on src/pages/RegisterToken/hooks/useRegisterTokenForm.ts:
```diff
@@ -103,7 +104,7 @@ export const useRegisterTokenForm = () => {
       await switchNetwork(NetworkChainIdMap[transaction.chainID]);
       await sleep(1000);
-      const { nonce } = accountSelector(sdkDappStore.getState());
+      const { nonce } = accountSelector(getState());
```
dynamic nonce
add comment why sleep

arhtudormorar commented on src/pages/Sign/Sign.tsx:
```diff
@@ -22,12 +23,14 @@ import { useValidateAndSignTxs } from './hooks';
 
-  const { address } = useGetAccount();
+  const { address } = useGetAccountInfo();
+  const network = networkSelector(getState());
```
Same

arhtudormorar commented on src/pages/SovereignTransfer/helpers/getSovereignTransferTransaction.ts:
```diff
@@ -24,7 +26,7 @@ export const getSovereignTransferTransaction = ({
 }) => {
   const egldLabel = getEgldLabel();
-  const { WEGLDid = '' } = getCurrentNetwork();
+  const { WEGLDid } = getCurrentNetwork();
```
Not reactive. Same goes for EGD label

razvantomegea replied:
Not meant to be reactive here. It is a function, not a hook or component

arhtudormorar commented on src/providers/PemProvider.ts:
```diff
@@ -0,0 +1,200 @@
+    // return ProviderTypeEnum.custom;
+  }
+
+  async signTransaction(transaction: Transaction) {
```
Seems similar to key store provider

arhtudormorar commented on src/redux/endpoints/faucet.endpoint.ts:
```diff
@@ -67,7 +63,7 @@ const faucetEndpoints = RootApi.injectEndpoints({
         }
 
-        const { decimals } = tokenData.data as PartialTokenType;
+        const { decimals } = tokenData.data as { decimals: number };
```
No need to cast

razvantomegea committed:
Refactor project structure and improve component functionality

- Removed the `PrivateKeyCheckWrapper` component and its related files to streamline the codebase.
- Updated `package.json` scripts to eliminate unnecessary storage configuration steps.
- Refactored various components to replace `modalTitle` with `panelTitle` for consistency in naming conventions.
- Introduced a new `readFromClipboard` helper function for improved clipboard handling.
- Enhanced transaction handling in hooks by integrating `sendAndTrackTransactions` for better transaction management.
- Cleaned up unused imports and adjusted routes for improved organization and maintainability.

razvantomegea committed:
Update SDK dependencies and refactor components for improved functionality

- Replaced `@multiversx/sdk-dapp-core-ui` with `@multiversx/sdk-dapp-ui` in multiple components to align with the latest SDK structure.
- Updated `package.json` to reflect the new version of `@multiversx/sdk-dapp-ui`.
- Removed the `BatchTransactionsContextProvider` component and its related files to simplify the codebase.
- Cleaned up imports in `CopyButton`, `ExplorerLink`, and `FormatAmount` components for consistency and clarity.

razvantomegea committed:
Add package-lock.json and update SDK dependencies in package.json

- Introduced a new `package-lock.json` to manage dependencies more effectively.
- Updated several SDK dependencies in `package.json`, including `@multiversx/sdk-core`, `@multiversx/sdk-dapp`, and `axios` to their latest versions for improved functionality.
- Refactored import paths in `FormatAmount` component for consistency.
- Cleaned up unused exports in `sdkDappUtils` and removed the `BatchTransactionsContextProvider` export from the wrappers.

razvantomegea committed:
Refactor AuthRedirectWrapper and introduce FeaturePageLayout component

- Removed unnecessary navigation logic from `AuthRedirectWrapper` to streamline authentication handling.
- Added a new `FeaturePageLayout` component for consistent layout structure across feature pages.
- Updated `SignMessage` component to utilize `FeaturePageLayout`, enhancing its structure and readability.

razvantomegea committed:
Refactor Faucet components to utilize FeaturePageLayout for consistent structure

- Updated `Faucet`, `FaucetError`, `FaucetScreen`, and `FaucetSuccess` components to use the new `FeaturePageLayout` for improved layout consistency.
- Removed redundant div wrappers and integrated titles directly into the `FeaturePageLayout`.
- Enhanced readability and maintainability of the components by streamlining their structure.

razvantomegea committed:
Enhance transaction handling in sendAndTrackTransactions hook

- Introduced support for both single and batch transactions by adding a new `signTxs` function to handle transaction signing.
- Integrated `getAccountProvider` to streamline provider access for signing transactions.
- Updated the transaction flow to ensure signed transactions are sent correctly, improving overall transaction management.

razvantomegea committed:
Merge branch 'development' into rt/feature/sdk-dapp-v5

razvantomegea committed:
Add start-sovereign-test script and update configurations for sovereign test network

- Introduced a new script `start-sovereign-test` in `package.json` to facilitate testing on the sovereign test network.
- Updated `config.sovereign-test.ts` to include new properties: `faucet`, `hasRegisterToken`, and `hasSovereignTransfer` for enhanced network configuration.
- Refactored `Account` and `RegisterToken` components to utilize the updated network configuration, improving state management and functionality.

razvantomegea committed:
Refactor index and header components to integrate network handling

- Removed unused imports and integrated `getCurrentNetwork` to dynamically set the active network in `index.tsx`.
- Updated `Header` component to include `NetworkSwitcher` for improved network selection, removing the previous active network display.
- Added `initializeNetwork` export in `sdkDapp.actions.ts` and included `GAS_PER_DATA_BYTE` in `sdkDapp.constants.ts` for enhanced network configuration.

razvantomegea committed:
Implement SidePanel component and refactor Keystore login handling

- Introduced a new `SidePanel` component for improved UI structure and user experience.
- Replaced the `KeystoreLoginModal` with `KeystoreLoginPanel` for a more consistent panel-based approach.
- Updated `KeystoreProvider` to utilize the new `KeystoreLoginPanel`, enhancing the login flow.
- Refactored imports in various components to include the new `Dropdown`, `Modal`, and `PageState` components for better organization.
- Adjusted the `index.tsx` to include `defineCustomElements` for custom element definitions.

razvantomegea committed:
Add KeystorePanel component and refactor KeystoreLoginPanel

- Introduced a new `KeystorePanel` component for handling keystore file uploads and user authentication.
- Updated `KeystoreLoginPanel` to utilize `KeystorePanel`, simplifying the login process and improving code organization.
- Refactored imports in `index.ts` and `components/index.ts` to include the new `KeystorePanel`, enhancing modularity.
- Removed unused imports and streamlined the code in `KeystoreLoginPanel` for better readability and maintainability.

razvantomegea committed:
Add PemPanel and refactor PEM login handling

- Introduced a new `PemPanel` component for handling PEM file uploads and user authentication.
- Replaced the `PemLoginModal` with `PemLoginPanel` for a more consistent panel-based approach.
- Updated `PemProvider` to utilize the new `PemLoginPanel`, enhancing the login flow.
- Refactored imports in `index.ts` and `providers/index.ts` to include the new `PemPanel` and `PemLoginPanel`, improving modularity and organization.
- Removed the deprecated `PemLoginModal` to streamline the codebase.

razvantomegea committed:
Enhance Keystore and PEM login handling with new components and refactor

- Introduced `KeystoreLoginPanel` and `PemLoginPanel` for improved user authentication flows.
- Added `PanelWrapper` components for both Keystore and PEM, enhancing the UI structure.
- Implemented new utility functions for handling keystore and PEM file parsing.
- Updated `KeystoreProvider` and `PemProvider` to utilize the new login panels, streamlining the login process.
- Refactored imports in `index.ts` and `providers/index.ts` for better organization and modularity.
- Removed deprecated components to clean up the codebase.

razvantomegea committed:
Add AddressScreens component and integrate into KeystorePanel

razvantomegea committed:
Refactor AddressScreens and KeystorePanel components for improved address handling

- Updated AddressScreens to manage account selection and pagination more effectively.
- Integrated Redux for state management in AddressScreens, allowing for dynamic address updates.
- Refactored KeystorePanel to utilize the new AddressScreens component, enhancing the user experience during wallet access.
- Cleaned up imports and removed deprecated code to streamline the components.

razvantomegea committed:
Refactor AddressScreens and KeystorePanel components for improved functionality

- Enhanced AddressScreens to support pagination and added a new onPageChange prop for better address management.
- Removed Redux integration from AddressScreens to simplify state handling.
- Cleaned up KeystorePanel by removing unused elements and improving layout consistency.
- Streamlined imports and removed deprecated code across components for better maintainability.

razvantomegea committed:
Enhance KeystorePanel and PanelWrapper for improved styling and functionality

- Updated styles in KeystorePanel for better layout and user experience, including adjustments to input fields, buttons, and error messages.
- Simplified PanelWrapper by removing unnecessary overlay and panel styles, directly rendering the KeystorePanel for cleaner code.
- Improved accessibility and visual consistency across the component's UI elements.

razvantomegea committed:
Enhance Button component by adding WithStyleType for improved styling options

- Updated ButtonType interface to include WithStyleType, allowing for enhanced styling capabilities.
- Improved type definitions for better maintainability and flexibility in button styling.

razvantomegea committed:
Refactor AddressScreens to use MvxAddressTable and improve address selection handling

- Replaced MvxAddressSelect with MvxAddressTable for better address management.
- Updated AddressScreensPropsType to enforce onConfirmSelectedAddress as a required prop.
- Simplified account selection logic and improved loading state handling.
- Cleaned up unused comments and streamlined event handling for address selection and pagination.

razvantomegea committed:
Fix import order in PanelWrapper component for consistency

razvantomegea committed:
Enhance KeystorePanel and PanelWrapper to support address handling

- Added `needsAddress` prop to KeystorePanel and PanelWrapper for conditional address selection.
- Updated KeystoreLoginPanel to pass `needsAddress` option to the rendered panel, improving flexibility in address management.

razvantomegea committed:
Enhance KeystorePanel and related components for improved file handling and submission

- Updated KeystorePanel to support additional properties in the onSubmit callback, including keystoreFile, keystoreFileName, and addressIndex.
- Integrated saved keystore file handling in KeystorePanel, allowing users to submit existing keystore data.
- Refactored PanelWrapper to render children components directly, improving flexibility in panel content.
- Modified KeystoreLoginPanel and KeystoreProvider to accommodate new properties for better state management and user experience.

razvantomegea committed:
Refactor routing and component structure for improved navigation and organization

- Removed the Unlock component from the main App routing and integrated it as a child route under Home.
- Updated the ConnectButton to navigate directly to the Unlock route instead of using the UnlockPanelManager.
- Added Outlet to the Home component to support nested routing for child components.
- Enhanced route type definitions to allow for child routes, improving the overall routing structure.

razvantomegea committed:
Remove Modal component and related exports for code cleanup

- Deleted Modal component and its index file to streamline the component library.
- Updated index.ts to reflect the removal of Modal, enhancing overall organization.
- Refactored handleLogout function in Header component for improved readability by removing unnecessary useCallback.

razvantomegea committed:
Refactor AxiosInterceptor, PageState, and PemPanel components for improved code consistency

- Updated AxiosInterceptor to use PropsWithChildren for better type handling.
- Modified PageStatePropsType to simplify the description prop type.
- Refactored PemPanel to utilize useState directly instead of React.useState, enhancing readability.

razvantomegea committed:
Update dependencies, adjust port configuration, and refactor imports for improved organization

- Updated MultiversX SDK dependencies in package.json to use version ranges for better compatibility.
- Changed default port in vite.config.ts from 3000 to 3001 to avoid conflicts.
- Refactored import paths in Playwright test utilities to improve clarity and consistency.
- Removed unused AuthRedirectWrapper component and its exports to streamline the component library.
- Enhanced Button component to utilize a new dataTestId prop for better testing capabilities.
- Updated PostMessageListener and related helper functions to improve message handling and logging.
- Added new hooks and types for better integration with the MultiversX SDK.

razvantomegea committed:
Refactor SignMessageHook and related components for improved message signing functionality

- Removed unused signMessage case from useRedirectPathname hook.
- Enhanced SignMessageHook to directly sign messages from the URL parameters and reply to the dapp with the signed message.
- Updated mapSignedTransactions to handle signed transactions more effectively.
- Refactored useSignHookTransactions to return signed transactions and improved transaction handling logic.
- Cleaned up imports and adjusted types for better clarity and maintainability.

razvantomegea committed:
Refactor PostMessageListener for improved message handling and session management

- Cleaned up import statements for better organization and clarity.
- Enhanced message navigation by utilizing HooksPageEnum for routing.
- Improved handshake session management by updating the payload handling in finalizeHandshakeRequest.
- Streamlined event listener setup and teardown to ensure proper cleanup and prevent memory leaks.

razvantomegea committed:
Refactor Sign component and useValidateAndSignTxs hook for improved transaction handling

- Updated the Sign component to utilize signedTransactions instead of rawTxs for better clarity and accuracy in transaction management.
- Refactored useValidateAndSignTxs hook to streamline state management and enhance the validation and signing process.
- Cleaned up unused variables and improved the overall structure for better maintainability.

razvantomegea committed:
Update yarn.lock and refactor SignMessage and Unlock hooks for improved type handling and functionality

- Updated yarn.lock to reflect new package versions and resolved URLs for better dependency management.
- Refactored useSignMessageCompleted hook to replace ExtendedReplyWithPostMessageType with ReplyWithPostMessageType for improved type clarity.
- Enhanced useOnLoginHookRedirect to ensure signature is always a string, preventing potential undefined issues.

razvantomegea committed:
Refactor useRedirectPathname and SignMessage components for improved message handling

- Updated useRedirectPathname to accept props for better flexibility in handling redirect paths based on hook type.
- Enhanced HookValidationOutcome to utilize the updated useRedirectPathname with isHook prop.
- Refactored SignMessage component to handle message signing directly from the hook URL and improved error handling.
- Removed automatic signing execution from SignMessageHook, delegating signing to the SignMessage page for better control.

razvantomegea committed:
Update package dependencies and enhance transaction handling in hooks

- Updated MultiversX SDK dependencies in package.json to use version ranges for better compatibility.
- Improved error handling in useValidateAndSignTxs hook by validating callback URLs and managing navigation based on origin.
- Refactored useSignMessageCompleted hook to simplify reply data structure.
- Ensured signature handling in useOnLoginHookRedirect is robust by providing a default value.

arhtudormorar commented on playwright-tests/utils/login.ts:
```diff
@@ -1,7 +1,7 @@
 import { Page } from '@playwright/test';
 import { confirmPem } from './confirmPem';
+import { LoginFilesEnum } from './enums';
```
FileLoginEnum

razvantomegea replied:
Incorrect.

arhtudormorar commented on src/components/AddressScreens/AddressScreens.tsx:
```diff
@@ -0,0 +1,118 @@
+import type { AccessWalletType } from '../../providers/Keystore/accessWallet';
+import { getKeystoreAddresses } from '../../providers/Keystore/getKeystoreAddresses';
+
+export interface IAccount {
```
use interfaces from skd-dapp-ui ?

arhtudormorar commented on src/components/AddressScreens/AddressScreens.tsx:
```diff
@@ -0,0 +1,118 @@
+  }, [loadAccounts, startIndex]);
+
+  const handleAccessWallet = useCallback(() => {
+    console.log('handleAccessWallet', selectedIndex);
```
logs

arhtudormorar commented on src/components/KeystorePanel/KeystorePanel.tsx:
```diff
@@ -0,0 +1,306 @@
+import { accessWallet } from '../../providers/Keystore/accessWallet';
+import { parseKeystoreJSON } from '../../providers/Keystore/parseKeystoreJSON';
+
+const styles = {
```
styles in local .scss file ?

arhtudormorar commented on src/components/KeystorePanel/KeystorePanel.tsx:
```diff
@@ -0,0 +1,306 @@
+  }
+};
+
+interface KeystorePanelProps {
```
IKeystorePanelProps || KeystorePanelPropsType

arhtudormorar commented on src/components/KeystorePanel/KeystorePanel.tsx:
```diff
@@ -0,0 +1,306 @@
+  const [savedFileContent, setSavedFileContent] = useState<string | null>(null);
+
+  useEffect(() => {
+    if (savedKeystoreFile) {
```
early return

arhtudormorar commented on src/components/KeystorePanel/KeystorePanel.tsx:
```diff
@@ -0,0 +1,306 @@
+      style={styles.container}
+      data-testid={DataTestIdsEnum.keystoreLoginPanel}
+    >
+      <form onSubmit={handleSubmit} style={styles.form}>
```
use formik

arhtudormorar commented on src/components/PanelWrapper/PanelWrapper.tsx:
```diff
@@ -0,0 +1,51 @@
+}: PanelWrapperProps) => {
+  const panelContent = (
+    <div
+      style={{
```
better scss

arhtudormorar commented on src/components/PemPanel/PemPanel.tsx:
```diff
@@ -0,0 +1,174 @@
+import { DataTestIdsEnum } from 'localConstants/dataTestIds.enum';
+import { parsePem } from 'providers/Pem/parsePem';
+
+const styles = {
```
scss

arhtudormorar commented on src/components/PemPanel/PemPanel.tsx:
```diff
@@ -0,0 +1,174 @@
+  }
+};
+
+interface PemPanelProps {
```
PemPanelPropsType | IPemPanelProps

arhtudormorar commented on src/components/PemPanel/PemPanel.tsx:
```diff
@@ -0,0 +1,174 @@
+
+  return (
+    <div style={styles.container} data-testid={DataTestIdsEnum.pemLoginPanel}>
+      <form onSubmit={handleSubmit} style={styles.form}>
```
use formik

arhtudormorar commented on src/components/SidePanel/SidePanel.tsx:
```diff
@@ -0,0 +1,47 @@
+import { MvxSidePanel } from '@multiversx/sdk-dapp-ui/react';
+
+// Define the component props based on the MvxSidePanel interface
+export interface SidePanelProps {
```
use naming convention

arhtudormorar commented on src/components/Utilities/components/PostMessageListener/helpers/buildTransactionsQueryString.ts:
```diff
@@ -8,6 +8,10 @@ export const buildTransactionsQueryString = ({
   transactions: Transaction[];
   callbackUrl?: string;
 }): string => {
+  console.log('[buildTransactionsQueryString] Input:', {
```
log

arhtudormorar commented on src/components/Utilities/components/PostMessageListener/helpers/buildTransactionsQueryString.ts:
```diff
@@ -24,7 +28,13 @@ export const buildTransactionsQueryString = ({
+    }
   });
+
+  console.log('[buildTransactionsQueryString] Result:', result);
```
log

arhtudormorar commented on src/hooks/useLogout/useLogout.ts:
```diff
@@ -52,14 +52,8 @@ export const useLogout = () => {
-      shouldAttemptReLogin,
-      options
-    );
+  return async () => {
```
return function direcly?

arhtudormorar commented on src/hooks/useRedirectPathname/useRedirectPathname.ts:
```diff
@@ -10,9 +10,10 @@ import { routeNames } from 'routes';
 export interface UseRedirectPathnameProps {
```
use convention

arhtudormorar commented on src/hooks/useReplyWithCancelled/useReplyWithCancelled.ts:
```diff
@@ -27,7 +29,7 @@ export const useReplyWithCancelled = (debugProps?: { caller: string }) => {
     if (props.shouldResetHook) {
       dispatch(resetHook({ wasCancelled: true }));
-      removeAllTransactions();
+      clearCompletedTransactions();
```
is this still needed?

arhtudormorar commented on src/index.tsx:
```diff
@@ -1,7 +1,50 @@
+  name: string;
+  type: string;
+  iconUrl: string;
+  constructor: (options?: any) => Promise<any>;
```
see interface from template

arhtudormorar commented on src/index.tsx:
```diff
@@ -1,7 +1,50 @@
+import { KeystoreProvider } from './providers/Keystore/KeystoreProvider';
+import { PemProvider } from './providers/Pem/PemProvider';
+
+interface ICustomProvider {
```
interface could come from sdk-dapp

arhtudormorar commented on src/index.tsx:
```diff
@@ -13,9 +56,11 @@ async function start() {
+  initApp(config).then(() => {
+    const container = document.getElementById('root');
+    const root = createRoot(container as HTMLElement);
+    root.render((<App />) as any);
```
no need as any

arhtudormorar commented on src/pages/Faucet/components/FaucetError/FaucetError.tsx:
```diff
@@ -1,25 +1,22 @@
-      {message}
-    </p>
-  </div>
+  <FeaturePageLayout title='Failed'>
```
why not wrap inside routes? or use layout settings

arhtudormorar commented on src/pages/Faucet/components/FaucetScreen/FaucetScreen.tsx:
```diff
@@ -33,34 +33,30 @@ export const FaucetScreen = ({
-        Request Tokens
-      </Button>
-    </div>
+    <FeaturePageLayout title={`${egldLabel} Faucet`}>
```
same

arhtudormorar commented on src/pages/Faucet/components/FaucetSuccess/FaucetSuccess.tsx:
```diff
@@ -1,25 +1,22 @@
-      {token} have been sent to your address.
-    </p>
-  </div>
+  <FeaturePageLayout title='Success'>
```
same

arhtudormorar commented on src/pages/Hook/SignMessageHook/SignMessageHook.tsx:
```diff
@@ -1,38 +1,49 @@
+  // Removed automatic sign execution; signing will be handled in SignMessage page
+
   useEffect(() => {
+    console.log('data', data);
```
logs

arhtudormorar commented on src/pages/IssueCollection/IssueCollection.tsx:
```diff
@@ -1,10 +1,8 @@
-    <h2 className='text-2xl font-bold p-2 mb-2 text-center'>
-      Issue Collection
-    </h2>
+  <FeaturePageLayout title='Issue Collection'>
```
same

arhtudormorar commented on src/pages/IssueNft/IssueNft.tsx:
```diff
@@ -1,8 +1,8 @@
 export const IssueNft = () => (
-  <div className='flex flex-col p-6 max-w-2xl w-full bg-white shadow-md rounded h-full'>
-    <h2 className='text-2xl font-bold p-2 mb-2 text-center'>Create NFT</h2>
+  <FeaturePageLayout title='Create NFT'>
```
same

arhtudormorar commented on src/pages/IssueToken/IssueToken.tsx:
```diff
@@ -1,8 +1,8 @@
 export const IssueToken = () => (
-  <div className='flex flex-col p-6 max-w-2xl w-full bg-white shadow-md rounded h-full'>
-    <h2 className='text-2xl font-bold p-2 mb-2 text-center'>Issue Token</h2>
+  <FeaturePageLayout title='Issue Token'>
```
same

arhtudormorar commented on src/pages/RegisterToken/RegisterToken.tsx:
```diff
@@ -1,22 +1,20 @@
-      <h2 className='text-2xl font-bold p-2 mb-2 text-center'>
-        Register Sovereign Token
-      </h2>
+    <FeaturePageLayout title='Register Sovereign Token'>
```
same

arhtudormorar commented on src/pages/Send/Send.tsx:
```diff
@@ -1,8 +1,8 @@
 export const Send = () => (
-  <div className='flex flex-col p-6 max-w-2xl w-full bg-white shadow-md rounded h-full'>
-    <h2 className='text-2xl font-bold p-2 mb-2 text-center'>Send</h2>
+  <FeaturePageLayout title='Send'>
```
same

arhtudormorar commented on src/pages/Sign/hooks/useSignHookTransactions.ts:
```diff
@@ -81,61 +70,45 @@ export const useSignHookTransactions = () => {
-
-      if (!batchSessionId) {
+      // Send as batch transactions
+      const sentTransactions = await txManager.send([signedTransactions]);
```
why send as batch?

arhtudormorar commented on src/pages/SignMessage/SignMessage.tsx:
```diff
@@ -1,159 +1,193 @@
-      className='flex flex-col p-6 max-w-2xl w-full bg-white shadow-md rounded h-full'
-      data-testid={DataTestIdsEnum.signMessagePage}
-    >
+    <FeaturePageLayout title='Sign Message'>
```
same

arhtudormorar commented on src/pages/SovereignTransfer/SovereignTransfer.tsx:
```diff
@@ -12,11 +13,8 @@ export const SovereignTransfer = () => {
-      <h2 className='text-2xl font-bold p-2 mb-2 text-center'>
-        Sovereign Transfer
-      </h2>
+    <FeaturePageLayout title='Sovereign Transfer'>
```
same

arhtudormorar commented on src/pages/Unlock/Unlock.tsx:
```diff
@@ -1,94 +1,35 @@
+  const { isLoggedIn } = useGetLoginInfo();
+
+  const unlockPanelManager = UnlockPanelManager.init({
+    loginHandler: () => {
```
loginHandler: onUnlockRedirect

arhtudormorar commented on src/providers/Keystore/KeystoreLoginPanel.tsx:
```diff
@@ -0,0 +1,127 @@
+import { KeystorePanel } from 'components/KeystorePanel';
+import { PanelWrapper } from 'components/PanelWrapper';
+
+interface KeystoreLoginPanelState {
```
IKeystoreLoginPanelState

arhtudormorar commented on src/providers/Keystore/KeystoreLoginPanel.tsx:
```diff
@@ -0,0 +1,127 @@
+interface KeystoreLoginPanelState {
+  root: Root;
+  isOpen: boolean;
+  resolve:
```
resolvedValue

arhtudormorar commented on src/providers/Keystore/KeystoreProvider.ts:
```diff
@@ -0,0 +1,222 @@
+      const savedKeystoreFile = state.account.keystoreFile;
+      const savedKeystoreFileName = state.account.keystoreFileName;
+
+      const { privateKey: userPrivateKey } = await this.panel.showPanel({
```
should not be trough an event

arhtudormorar commented on src/providers/Pem/PemLoginPanel.tsx:
```diff
@@ -0,0 +1,97 @@
+import PanelWrapper from 'components/PanelWrapper/PanelWrapper';
+import { PemPanel } from 'components/PemPanel';
+
+interface PemLoginPanelState {
```
naming convention

arhtudormorar commented on src/providers/Pem/PemLoginPanel.tsx:
```diff
@@ -0,0 +1,97 @@
+interface PemLoginPanelState {
+  root: Root;
+  isOpen: boolean;
+  resolve: ((value: { privateKey: string; address: string }) => void) | null;
```
resolvedValue

arhtudormorar commented on src/providers/Pem/PemProvider.ts:
```diff
@@ -0,0 +1,198 @@
+
+let privateKey = '';
+
+export class PemProvider implements IProvider {
```
why do we have 2 provider classes ?

arhtudormorar commented on src/redux/endpoints/faucet.endpoint.ts:
```diff
@@ -1,15 +1,17 @@
+// All @multiversx/sdk-dapp-form imports removed for migration. Add equivalents from sdk-dapp-utils or custom logic if needed.
```
add TODO

arhtudormorar commented on src/routes/routes.tsx:
```diff
@@ -26,23 +24,29 @@ import {
-export interface RouteWithTitleType extends RouteType {
+export interface RouteType {
+  authenticatedRoute?: boolean;
+  path: RouteNamesEnum | HooksPageEnum | CreateRecoverRoutesEnum | string;
```
seems strange, thre should be a better way. TBD

arhtudormorar commented on src/routes/routes.tsx:
```diff
@@ -26,23 +24,29 @@ import {
 const routesObject: Record<
-  RouteNamesEnum | HooksPageEnum | CreateRecoverRoutesEnum,
-  RouteWithTitleType
+  RouteNamesEnum | HooksPageEnum | CreateRecoverRoutesEnum | string,
```
same

arhtudormorar commented on src/routes/routes.tsx:
```diff
@@ -64,83 +68,47 @@ const routesObject: Record<
-        <RegisterToken />
-      </PrivateKeyCheckWrapper>
-    )
+    component: () => <RegisterToken />
```
use layout at this level

arhtudormorar commented on src/types/withClassName.types.ts:
```diff
@@ -0,0 +1,8 @@
+  'data-testid'?: string;
+};
+
+export type WithStyleType = {
```
is this needed?

arhtudormorar commented on src/utils/app/getIsInWebview.ts:
```diff
@@ -1,15 +1,20 @@
     const state: RootState = store.getState();
-    const providerType = getAccountProviderType();
-    const isExternalProvider = providerType === LoginMethodsEnum.extra;
+    const provider = getAccountProvider();
```
getIsInWebview exists in sdk-dapp or webview provider
is it needed here?

razvantomegea replied:
It doesn't

arhtudormorar commented on src/utils/app/provider.ts:
```diff
@@ -1,31 +1,41 @@
 
-export const provider: IDappProvider = {
-  init: async () => {
+class CustomProvider implements IProvider {
```
is this needed here?
For all providers I think it was better before as object

arhtudormorar commented on src/utils/app/provider.ts:
```diff
@@ -35,65 +45,81 @@ export const provider: IDappProvider = {
+
+  getType(): ProviderType {
+    // TODO: Add custom provider type
+    return (ProviderTypeEnum as any).custom;
```
why do we define a custom provider?

razvantomegea committed:
Refactor provider structure and enhance type handling

- Introduced IFileProvider and IFileProviderOptions interfaces for better type management in file providers.
- Updated KeystoreProvider and PemProvider constructors to utilize the new IFileProviderOptions interface.
- Refactored the provider initialization in index.tsx to streamline provider management.
- Simplified the return value of useIsWebProvider hook for improved clarity.
- Added CLAUDE.md to .gitignore for better file management.

razvantomegea committed:
Refactor AddressScreens, KeystorePanel, and provider components for improved clarity and structure

- Removed unnecessary console logs from AddressScreens for cleaner code.
- Renamed KeystorePanelProps to IKeystorePanelProps for consistency with interface naming conventions.
- Introduced FileProvider as a base class for KeystoreProvider and PemProvider to streamline provider functionality.
- Updated provider methods to enhance type handling and reduce redundancy.
- Improved the handling of login and reauthentication processes in both KeystoreProvider and PemProvider.

razvantomegea committed:
Refactor FileLoginPanel and Button components for improved clarity and functionality

- Introduced FileLoginPanel component for file upload handling with enhanced styling and structure.
- Updated Button component to simplify type definitions and improve consistency.
- Adjusted imports in various components to streamline the codebase and enhance maintainability.
- Removed unnecessary interface definitions and improved type handling across components.

razvantomegea committed:
Refactor KeystorePanel and PemPanel to utilize FileLoginPanel for improved structure and reusability

- Replaced form handling in KeystorePanel and PemPanel with FileLoginPanel for consistent file upload functionality.
- Updated interface names to follow naming conventions, changing KeystorePanelProps to IKeystorePanelProps and PemPanelProps to IPemPanelProps.
- Streamlined component structure by removing redundant styles and form elements, enhancing maintainability.

razvantomegea committed:
Enhance FileLoginPanel integration and validation for improved file handling

- Refactored FileLoginPanel to utilize Formik for form management, enhancing validation and error handling.
- Updated KeystorePanel and PemPanel to leverage the new FileLoginPanel structure, improving consistency in file upload processes.
- Introduced fileLoginFormSchema for robust validation of file and password inputs.
- Adjusted component props to support initial values and improved type definitions for better clarity and maintainability.

razvantomegea committed:
Refactor FileLoginPanel and PostMessageListener for improved clarity and functionality

- Updated FileLoginPanel props to enhance type definitions for better clarity.
- Reintroduced the getIsInWebview import in PostMessageListener for improved functionality.
- Ensured proper formatting in fileLoginFormSchema by adding a newline at the end of the file.

razvantomegea committed:
Update import path for Address utility to enhance modularity and maintainability

razvantomegea committed:
Refactor multiple components to remove FeaturePageLayout for improved simplicity

- Updated Faucet, IssueCollection, IssueNft, IssueToken, RegisterToken, Send, and SovereignTransfer components to eliminate the FeaturePageLayout wrapper, streamlining the component structure.
- Adjusted routes to include FeaturePageLayout where necessary, ensuring consistent layout usage across the application.
- Enhanced the SignMessage component layout while maintaining its functionality.

razvantomegea committed:
Implement BaseFileLoginPanel as a reusable component for login panels

- Introduced BaseFileLoginPanel class to standardize the structure and functionality of login panels.
- Refactored KeystoreLoginPanel and PemLoginPanel to extend BaseFileLoginPanel, enhancing code reusability and maintainability.
- Updated interfaces for login options and return values to align with the new base class structure.
- Streamlined rendering logic and state management across login panels for improved clarity and functionality.

razvantomegea committed:
Update FeaturePageLayout title attributes for consistency

- Changed double quotes to single quotes for title attributes in the Register Token and Faucet routes to maintain uniformity in code style.

razvantomegea committed:
Refactor KeystorePanel for improved error handling and code consistency

- Added early return in useEffect to handle cases where savedKeystoreFile is null, enhancing error management.
- Standardized string quotes in props and placeholders for consistency across the component.
- Improved readability by ensuring consistent formatting in input fields and labels.

razvantomegea committed:
Standardize string quotes in KeystorePanel for consistency

- Updated all string quotes in props and placeholders from double quotes to single quotes to maintain uniformity across the component.
- Enhanced readability by ensuring consistent formatting in input fields and labels.

razvantomegea committed:
Refactor AddressScreens component to improve type definitions and data handling

- Updated the state management to use IndexedAccountType instead of the previous IAccount interface for better alignment with the new data structure.
- Changed the accountScreenData type from IAccountScreenData to IAddressTableData to enhance consistency with the updated types.
- Removed deprecated interface definitions to streamline the code and improve maintainability.

razvantomegea committed:
Enhance FileLoginPanel and PemPanel for improved error handling and integration

- Added an error prop to FileLoginPanel for better error display management.
- Updated PemPanel to pass the error prop to FileLoginPanel, ensuring consistent error handling.
- Simplified password validation in fileLoginFormSchema for clarity and maintainability.

razvantomegea committed:
Remove unnecessary click action for the Connect button in loginWithKeystore function to streamline the login process.

arhtudormorar commented on src/components/AddressScreens/AddressScreens.tsx:
```diff
@@ -0,0 +1,105 @@
+  return (
+    <MvxAddressTable
+      className={className}
+      selectedIndex={selectedIndex ?? undefined}
```
?? undefined seems a bit strange

razvantomegea committed:
Refactor jest-puppeteer configuration and enhance testing utilities

- Simplified the jest-puppeteer.config.js by removing redundant functions and integrating a new module for Chrome executable path resolution.
- Introduced jest-puppeteer.localhost.js for local testing with a configurable base URL.
- Updated package.json to include a new script for testing with the localhost configuration.
- Enhanced Vite configuration to use port 3002 for local development.
- Added utility functions for deep querying elements by data-testid in Puppeteer tests, improving test reliability and maintainability.
- Refactored various test files to utilize the new deep query utility, ensuring consistent element selection across tests.

razvantomegea committed:
Enhance testing utilities and improve error handling in transaction processes

- Added mock responses for blocks and latest blocks in server mocks to facilitate testing.
- Refactored sendAndTrackTransactions function to include error handling, ensuring robustness during transaction processing.
- Created a new helper function for navigating to the Issue Collection page, streamlining test code across multiple test files.
- Updated various Issue Collection tests to utilize the new navigation helper, improving code maintainability and readability.

razvantomegea committed:
Update dependencies in yarn.lock and modify GitHub workflows for testing

- Updated various package versions in yarn.lock to their latest releases, ensuring improved performance and security.
- Changed the runner for Playwright tests from 'ubuntu-latest' to 'runner-dapp' for better compatibility with the testing environment.
- Adjusted the indentation in the Playwright workflow for consistency.
- Updated the runner for pre-merge e2e tests to 'runner-dapp' to align with the new testing strategy.

razvantomegea committed:
Enhance server mocks and update test utilities

- Added a new transactions mock to improve testing capabilities for transaction-related functionalities.
- Updated server mock handlers to include the new transactions endpoint for better simulation of API responses.
- Modified test files to streamline button click actions, enhancing readability and maintainability.
- Adjusted login utility functions to improve delay handling and ensure consistent user experience during tests.

razvantomegea committed:
Refactor transaction handling in tests and add new navigation helper

- Replaced expectElementToContainText with expectAndSignTransaction in multiple test files to streamline transaction verification.
- Introduced a new helper function, navigateToIssueNftPage, to simplify navigation in NFT-related tests.
- Added sidePanel to DataTestIdsEnum for improved element targeting during transaction signing.
- Updated various test files to utilize the new helper and transaction handling method, enhancing code maintainability and readability.

razvantomegea committed:
Add navigation helper for Issue Token tests and refactor test cases

- Introduced a new helper function, navigateToIssueTokenPage, to streamline navigation in Issue Token tests.
- Refactored existing test cases to utilize the new helper, improving code maintainability and readability.
- Removed redundant navigation code from multiple test files, enhancing clarity and reducing duplication.

razvantomegea committed:
Add navigation helper for Register Token tests and refactor test cases

- Introduced a new helper function, navigateToRegisterTokenPage, to streamline navigation in Register Token tests.
- Refactored existing test cases in RegisterEsdtToken.e2e.ts and RegisterNftToken.e2e.ts to utilize the new helper, improving code maintainability and readability.
- Removed redundant navigation code from multiple test files, enhancing clarity and reducing duplication.

razvantomegea committed:
Refactor import statements in Register Token tests

- Simplified import statements in RegisterEsdtToken.e2e.ts and RegisterNftToken.e2e.ts by removing unnecessary line breaks, enhancing code readability.

razvantomegea committed:
Refactor loginWithKeystore and related tests to remove skipLoggedInCheck parameter

- Removed the skipLoggedInCheck parameter from loginWithKeystore and loginWithPem functions to simplify the login process.
- Updated multiple test files to reflect this change, ensuring consistent login behavior across tests.
- Enhanced element selection in SovereignTransfer.e2e.ts for improved clarity and maintainability.

razvantomegea committed:
Update dependencies and switch from Yarn to PNPM for package management

- Added new dependencies: lodash and qs, along with their type definitions.
- Updated several devDependencies to their latest versions for improved performance and compatibility.
- Introduced pnpm.lock file and removed yarn.lock to standardize package management.
- Modified GitHub workflow to use PNPM instead of Yarn for installing dependencies and running tests.

razvantomegea committed:
Update GitHub workflows to use 'ubuntu-latest' runner

- Changed the runner for Playwright and pre-merge e2e tests from 'runner-dapp' to 'ubuntu-latest' for improved compatibility and consistency across testing environments.

razvantomegea committed:
Update babel-preset-current-node-syntax to version 1.1.1 and adjust lodash import in getKeystoreAddresses

- Upgraded babel-preset-current-node-syntax from 1.1.0 to 1.1.1, updating peer dependencies for compatibility.
- Refactored lodash import in getKeystoreAddresses.ts to use the specific 'range' function, improving performance and reducing bundle size.

razvantomegea committed:
Refactor HRP setup and update network configuration

- Removed the useSetupHrp hook to streamline HRP configuration.
- Updated the sovereign test configuration to set the default address HRP to 'vibe' and modified network details for VibeChain.
- Added the setProviderType function to FileProvider for improved provider management.
- Specified provider types in KeystoreProvider and PemProvider classes for better clarity.

razvantomegea committed:
Update dependencies and refactor workflows for improved compatibility

- Updated pnpm-lock.yaml to reflect new versions for several packages, including @multiversx/sdk-dapp and @floating-ui/core.
- Refactored GitHub workflows to fix permissions and use the latest checkout action, enhancing deployment consistency across environments.
- Updated mock data for VibeChain, including changes to wallet addresses and API endpoints, ensuring alignment with the new network configuration.
- Added new test utilities for better handling of URL matching and element selection in Puppeteer tests, improving test reliability and maintainability.

razvantomegea committed:
Update network configuration and enhance test utilities

- Changed the erd_chain_id in networkConfig.ts from 'S' to 'V' to align with the new network settings.
- Added new data test IDs in dataTestIds.enum.ts for improved element targeting in tests.
- Removed unnecessary console log in HookValidationOutcome.tsx to clean up the code.
- Updated various end-to-end tests to utilize the new expectAndSignTransaction method for better transaction verification.
- Introduced new tests for handling invalid sign messages, ensuring robust error handling in the SignMessageHook.
- Refactored existing tests to improve clarity and maintainability, including the addition of keystoreAccount in several test files for consistency.

razvantomegea committed:
Update test configurations and improve gas price handling

- Updated signerAddress in multiple end-to-end test files to include '@' prefix for consistency.
- Adjusted gasPrice values across various tests to a more precise format.
- Enhanced the expectAndSignTransaction utility to reflect updated gas price and limit formatting in transaction requests.
- Improved clarity in transaction success messages by changing text to indicate 'Sent VIBE'.

razvantomegea committed:
Refactor end-to-end tests to remove usdAmount and ensure consistent transaction handling

- Removed usdAmount from multiple end-to-end test files to streamline transaction data.
- Updated expectAndSignTransaction utility to reflect changes in transaction structure.
- Ensured consistent gas price and limit formatting across various tests for improved reliability.

razvantomegea committed:
Refactor transaction creation in forms to use await for asynchronous operations

- Updated transaction creation in multiple form hooks to use await for better handling of asynchronous operations.
- Changed public key generation method in getKeysFromMnemonic to ensure consistency in address formatting.

razvantomegea committed:
Update package.json and pnpm-lock.yaml for dependency management

- Added a new script "install-verdaccio" to package.json for managing @multiversx/sdk-dapp installation.
- Updated versions of @multiversx/sdk-dapp-ui and @multiversx/sdk-web-wallet-cross-window-provider in pnpm-lock.yaml to ensure compatibility with the latest changes.
- Incremented versions for several dependencies, including @bufbuild/protobuf and chalk, to maintain up-to-date package management.

razvantomegea committed:
Enhance transaction data handling in createNewTransactionsFromRaw

- Updated the createNewTransactionsFromRaw function to encode transaction data in base64 format before creating new transaction objects.
- Improved the mapping logic to ensure that the transaction data is consistently processed, enhancing data integrity.

razvantomegea committed:
Enhance transaction handling and update test utilities

- Added new data test IDs in dataTestIds.enum.ts for improved element targeting in tests.
- Refactored expectAndSignTransaction utility to support additional parameters for better transaction data handling.
- Updated end-to-end tests to utilize new transaction structure and ensure consistent messaging for transaction processing.
- Improved UI feedback in Sign component to indicate transaction status with appropriate icons and messages.

razvantomegea committed:
Update mock data and enhance address validation for VibeChain

- Increased Jest timeout for Puppeteer tests from 60s to 120s to accommodate longer test execution times.
- Updated mock addresses in faucetSettings, issueContract, transactions, and emptyWallet to reflect the new VibeChain format.
- Replaced the deprecated addressIsErd function with addressIsHrp for better address validation, ensuring compatibility with the new HRP structure.
- Added comprehensive tests for addressIsHrp to validate various address formats and ensure robust error handling.

razvantomegea committed:
Update mock wallet address in emptyWallet.ts to reflect new VibeChain format

razvantomegea committed:
Update end-to-end tests for RegisterToken to reflect new contract address format

- Removed keystoreAccount reference and replaced it with the updated contract address in both RegisterEsdtToken and RegisterNftToken tests.
- Adjusted gas price and limit values for improved transaction handling.
- Enhanced test reliability by adding a sleep delay to accommodate page load times.

razvantomegea committed:
Update Jest Puppeteer configuration and enhance end-to-end tests for Send NFT

- Increased maxWorkers in jest-puppeteer.config.js to improve test execution efficiency.
- Refactored PrivateKeyCheckSend.e2e.ts to include additional utility functions and enhance transaction handling.
- Added sleep delays to accommodate page load times and ensure reliable test execution.
- Updated input handling for transaction data, including gas limit and receiver address validation.

razvantomegea committed:
Add skipValidation prop to FileLoginPanel and KeystorePanel for conditional validation

- Introduced skipValidation prop in FileLoginPanel to bypass validation schema when needed.
- Updated button disabled state logic in FileLoginPanel to account for skipValidation.
- Modified initialValues in KeystorePanel to conditionally set file based on savedFileContent and added skipValidation prop accordingly.

razvantomegea committed:
Update Jest Puppeteer configuration to disable test bail option

- Changed the bail option in jest-puppeteer.config.js from 1 to 0 to allow all tests to run regardless of failures, improving test coverage during execution.

razvantomegea committed:
Update signer address in PrivateKeyCheckSend.e2e.ts for improved test clarity

- Changed the signerAddress in the Send NFT tests from keystoreAccount.address to a specific identifier '@webteam' to enhance test readability and maintainability.

razvantomegea committed:
Update Jest Puppeteer configuration to adjust test execution parameters

- Changed the bail option in jest-puppeteer.config.js from 0 to 1 to stop test execution on the first failure, enhancing error visibility.
- Reduced maxWorkers from 4 to 2 to optimize resource usage during test runs.

razvantomegea committed:
Update Jest Puppeteer configuration to set maxWorkers to 1 for resource optimization

- Changed the maxWorkers setting in jest-puppeteer.config.js from 2 to 1 to further optimize resource usage during test execution.

razvantomegea committed:
Update transaction data in SovereignTransfer.e2e.ts for improved test accuracy

- Modified the data field in the SovereignTransfer end-to-end test to reflect updated transaction parameters, ensuring tests align with the latest contract specifications.

razvantomegea committed:
Update transaction data in SovereignTransfer.e2e.ts for test accuracy

- Modified the dataHighlight field in the SovereignTransfer end-to-end test to reflect the latest transaction parameters, ensuring alignment with current contract specifications.

arhtudormorar commented on package.json:
```diff
@@ -31,9 +31,11 @@
     "test": "jest",
-    "preview": "vite preview"
+    "preview": "vite preview",
+     "install-verdaccio": "pnpm remove @multiversx/sdk-dapp && pnpm install @multiversx/sdk-dapp@5.0.11 --registry http://localhost:4873"
```
@multiversx/sdk-dapp@latest ?

arhtudormorar commented on package.json:
```diff
@@ -44,20 +46,23 @@
-    "@multiversx/sdk-dapp-form": "2.0.2",
-    "@multiversx/sdk-js-web-wallet-io": "2.0.5",
-    "@multiversx/sdk-web-wallet-cross-window-provider": "3.1.0",
+    "@multiversx/sdk-core": "^14.x",
```
or 15

arhtudormorar commented on src/components/FileLoginPanel/FileLoginPanel.tsx:
```diff
@@ -0,0 +1,199 @@
+  skipValidation?: boolean;
+}
+
+export const FileLoginPanel = ({
```
to me it seems it has too many props but OK

arhtudormorar commented on src/components/KeystorePanel/KeystorePanel.tsx:
```diff
@@ -0,0 +1,240 @@
+
+  const handleConfirmSelectedAddress = useCallback(
+    (account: { index: number; address: string }) => {
+      if (!account || !keystoreData) return;
```
add eslint rule brackets for if

arhtudormorar commented on src/components/SidePanel/SidePanel.tsx:
```diff
@@ -0,0 +1,47 @@
+  onClose,
+  onBack
+}: ISidePanelProps) => {
+  const handleClose = useCallback(() => {
```
could be directly in TSX but OK

arhtudormorar commented on src/hooks/useRedirectPathname/useRedirectPathname.ts:
```diff
@@ -26,7 +27,9 @@ export const useRedirectPathname = () => {
         return routeNames.sign;
       case HooksEnum.signMessage:
-        return routeNames.signMessage;
+        return props?.isHook
```
add comment here why this logic interference

arhtudormorar commented on src/hooks/useSendTransactions/helpers/sendAndTrackTransactions.ts:
```diff
@@ -0,0 +1,49 @@
+  const provider = getAccountProvider();
+
+  // Support both single and batch transactions
+  const signTxs = async (txs: Transaction[] | Transaction[][]) => {
```
add return type => Promise...

arhtudormorar commented on src/index.tsx:
```diff
@@ -1,7 +1,44 @@
+    network: {
+      ...activeNetwork,
+      walletAddress:
+        activeNetwork.walletAddress || 'https://devnet-wallet.multiversx.com'
```
I think this is not needed

arhtudormorar commented on src/lib/sdkDapp/sdkDapp.actions.ts:
```diff
@@ -0,0 +1,3 @@
+export { createCustomToast } from '@multiversx/sdk-dapp/out/store/actions/toasts/toastsActions';
+export { initializeNetwork } from '@multiversx/sdk-dapp/out/store/actions/network/networkActions';
+export { setProviderType } from '@multiversx/sdk-dapp/out/store/actions/loginInfo/loginInfoActions';
```
whre is this used?

arhtudormorar commented on src/pages/Dashboard/components/Widget.tsx:
```diff
@@ -30,7 +30,7 @@ export const Widget = ({
   props = {}
 }: WidgetType) => {
-  const { isWebProvider } = useIsWebProvider();
+  const isWebProvider = useIsWebProvider();
```
what is useIsWebProvider ?

arhtudormorar commented on src/pages/Hook/SignMessageHook/SignMessageHook.tsx:
```diff
@@ -1,20 +1,22 @@
   const { pathname, search } = useLocation();
+  const { hookUrl } = useSelector(hookSelector);
 
   const data = useMemo(() => {
```
`  const data = useMemo(() => getSignMessageHookData(search.length > 0 ? search : hookUrl), [pathname]);`

arhtudormorar commented on src/pages/IssueNft/IssueNft.tsx:
```diff
@@ -1,8 +1,3 @@
 import { IssueNftForm } from './components';
```
export { IssueNftForm as IssueNft } from './components';

arhtudormorar commented on src/pages/IssueToken/IssueToken.tsx:
```diff
@@ -1,8 +1,3 @@
 import { IssueTokenForm } from './components';
```
same

arhtudormorar commented on src/pages/Send/Send.tsx:
```diff
@@ -1,8 +1,3 @@
 import { SendForm } from './components';
```
same

arhtudormorar commented on src/pages/Sign/Sign.tsx:
```diff
@@ -119,14 +126,58 @@ export const Sign = () => {
+          Processing Transaction
+        </h2>
+        <p className='text-gray-600'>
+          Your transaction is being processed. Please wait...
```
why not toast here?

arhtudormorar commented on src/pages/Sign/hooks/useValidateAndSignTxs.ts:
```diff
@@ -26,116 +19,71 @@ import { mapSignedTransactions } from '../helpers';
 
     if (isValidHook) {
-      replyWithSignedTransactions(transactions);
+      try {
```
use early return for if

arhtudormorar commented on src/routes/routes.tsx:
```diff
@@ -65,81 +69,77 @@ const routesObject: Record<
     title: 'Issue Token',
     component: () => (
-      <PrivateKeyCheckWrapper>
+      <FeaturePageLayout title='Issue Token'>
```
titles are dublicated
I think it was better with a new prop called `layout: LayoutsEnum.FeaturePage`
then a switch that wrapps the component with the correct layout and inserts the title as prop

razvantomegea replied:
Future feature

razvantomegea committed:
Update project configuration and improve component logic

- Added new entries to .gitignore for additional files and directories.
- Adjusted maxWorkers setting in jest-puppeteer.config.js to improve test performance based on headless mode.
- Updated package.json to install the latest version of @multiversx/sdk-dapp.
- Refactored FileProvider and KeystoreProvider to use FileProviderEnum for type consistency.
- Enhanced AddressScreens component state management by changing selectedIndex to use undefined instead of null.
- Improved buildWalletQueryString type definition for better clarity.
- Renamed useIsWebProvider to useIsWebviewProvider for better accuracy.
- Simplified several page components by directly exporting forms from their respective modules.

razvantomegea committed:
Refactor buildWalletQueryString for improved type clarity

- Enhanced the type definition of buildWalletQueryString to improve readability and maintainability by formatting the options parameter more clearly.

razvantomegea committed:
Refactor wallet address handling and introduce sign with redirect hook

- Removed default wallet address assignment in the network configuration to rely solely on the active network's wallet address.
- Added a new hook, useSignWithRedirect, to manage transaction signing with redirection logic.
- Updated useRedirectPathname to change the redirect path for the sign hook to the dashboard.
- Cleaned up imports by removing unused setProviderType references in multiple files.

razvantomegea committed:
Update pnpm-lock.yaml to reflect dependency version changes

- Updated '@multiversx/sdk-dapp' and '@multiversx/sdk-js-web-wallet-io' versions from 5.0.11 to 5.0.13 and 3.1.0 to 3.1.0 respectively.
- Updated '@swc/types' version from 0.1.23 to 0.1.24.
- Updated various dependencies to include 'sass@1.90.0' across multiple packages.
- Added new optional dependencies for '@parcel/watcher' across different platforms.
- Updated 'yaml' version from 2.8.0 to 2.8.1 for improved compatibility.

razvantomegea committed:
Update Jest Puppeteer configuration to set maxWorkers to 50% for improved resource management in headless mode

razvantomegea committed:
Update Jest Puppeteer configuration to set maxWorkers to 2 for improved performance in headless mode

razvantomegea committed:
Remove unused sleep function from useRegisterTokenForm hook to streamline transaction processing

arhtudormorar commented on package.json:
```diff
@@ -44,20 +46,23 @@
+    "@multiversx/sdk-web-wallet-cross-window-provider": "^3.x",
     "@reduxjs/toolkit": "1.9.1",
-    "axios": "1.6.5",
+    "axios": "1.7.4",
```
1.10.0 ?

arhtudormorar commented on src/hooks/useSignWithRedirect/useSignWithRedirect.ts:
```diff
@@ -0,0 +1,81 @@
+      console.error('Invalid hook');
+    }
+
+    if (senderAddresses.length > 1) {
```
senderAddresses.length > 1 --> extract to named variable

razvantomegea committed:
Enhance type definition for signTxs function in sendAndTrackTransactions helper

- Updated the signTxs function to explicitly define its return type as Promise<Transaction[] | Transaction[][]>, improving type clarity and maintainability.

arhtudormorar commented on src/pages/Sign/Sign.tsx:
```diff
@@ -22,40 +27,44 @@ import { useValidateAndSignTxs } from './hooks';
 export const Sign = () => {
   const { hookUrl } = useSelector(hookSelector);
-  const { loginMethod } = useGetLoginInfo();
+  const provider = getAccountProvider();
```
remove file ?

razvantomegea committed:
Refactor transaction signing process and enhance related hooks

- Removed the Sign page and its associated components to streamline the signing process.
- Introduced new helper functions for creating and preparing transactions from raw data.
- Updated the useSignWithRedirect hook to improve transaction validation and signing logic.
- Enhanced the useValidateAndSignTxs hook for better state management and error handling.
- Cleaned up imports and improved code organization across multiple files.

razvantomegea committed:
Update project version to 3.0.0 and enhance dependencies

- Bump project version from 2.0.2 to 3.0.0 in package.json.
- Update dependency specifications in package.json and pnpm-lock.yaml to use 'latest' versions for several MultiversX SDK packages.
- Remove 'docs/' from .gitignore to include documentation in version control.
- Add a new project brief document outlining the MultiversX Lite Wallet DApp's features and architecture.

razvantomegea committed:
Update pnpm-lock.yaml to reflect dependency version changes

- Updated versions for several MultiversX SDK packages, including '@multiversx/sdk-dapp-ui' from 0.0.23 to 0.0.24 and '@multiversx/sdk-js-web-wallet-io' remains at 3.1.0.
- Added new dependencies for '@storybook' packages, including '@storybook/addon-actions', '@storybook/addon-links', and others, to enhance the development environment.
- Updated various '@esbuild' packages to version 0.25.8 for improved compatibility and performance across multiple platforms.
- Included new optional dependencies and peer dependencies for better integration with testing libraries and build tools.

razvantomegea committed:
Update pnpm-lock.yaml and .gitignore for dependency management

- Updated several MultiversX SDK package versions in pnpm-lock.yaml, including '@multiversx/sdk-dapp' from 5.0.13 to 5.1.1 and '@multiversx/sdk-dapp-ui' from 0.0.24 to 0.0.25.
- Added new versions for various dependencies, including '@jridgewell/gen-mapping' and '@jridgewell/trace-mapping'.
- Included 'docs/' in .gitignore to exclude documentation from version control.

razvantomegea committed:
Refactor PanelWrapper and SidePanel components to remove unused onBack prop

- Removed the onBack prop from both PanelWrapper and SidePanel components to simplify their interfaces.
- Updated the KeystoreLoginPanel and PemLoginPanel to reflect the removal of the onBack prop, enhancing code clarity and maintainability.

razvantomegea committed:
Refactor BaseFileLoginPanel to simplify panel management and improve rendering logic

- Removed the internal state management and initialization logic from BaseFileLoginPanel, replacing it with a more straightforward approach for rendering and handling submissions.
- Updated KeystoreLoginPanel and PemLoginPanel to remove unnecessary constructors and instance destruction methods, enhancing code clarity.
- Cleaned up imports in expectAndSignTransaction utility for better organization.
